time-tree.js

```diff
@@ -0,0 +1,86 @@
+'use strict'
+const archy = require('archy')
+
+function TimeTree () {
+  this.root = null
+}
+
+TimeTree.prototype.search = function (filter) {
+  const seachNode = (node) => {
+    const isMe = filter(node)
+    if (!isMe) {
+      return node.nodes
+        .map(_ => seachNode(_))
+        .filter(_ => _ !== null)
+        .pop()
+    }
+    return node
+  }
+  return seachNode(this.root)
+}
+
+TimeTree.prototype.getParent = function (parent, nodeId) {
+  if (parent === null) {
+    return this.root
+  }
+
+  return this.search(node => node.label === parent)
+}
+
+TimeTree.prototype.getNode = function (nodeId) {
+  return this.search(node => node.id === nodeId)
```

Comment:
This is going to get pretty hot, and slow us during boot time. Can you create a Map where you store the ids and look them up instead?